cmd/ssh.go

```diff
@@ -192,16 +226,78 @@ func (cmd *SSHCmd) jumpContainer(ctx context.Context, devPodConfig *config.Confi
 	}
 
 	// tunnel to container
-	return tunnel.NewContainerTunnel(client, cmd.Proxy, log).Run(ctx, func(ctx context.Context, containerClient *ssh.Client) error {
-		// we have a connection to the container, make sure others can connect as well
-		unlockOnce.Do(client.Unlock)
+	return tunnel.NewContainerTunnel(client, cmd.Proxy, log).
+		Run(ctx, func(ctx context.Context, containerClient *ssh.Client) error {
+			// we have a connection to the container, make sure others can connect as well
+			unlockOnce.Do(client.Unlock)
+
+			// start ssh tunnel
+			return cmd.startTunnel(
+				ctx,
+				devPodConfig,
+				containerClient,
+				client.WorkspaceConfig().IDE.Name,
+				log,
+			)
+		})
+}
+
+func (cmd *SSHCmd) reverseForwardPorts(
+	ctx context.Context,
+	containerClient *ssh.Client,
+	log log.Logger,
+) error {
+	timeout := time.Duration(0)
+	if cmd.ForwardPortsTimeout != "" {
+		var err error
+		timeout, err = time.ParseDuration(cmd.ForwardPortsTimeout)
+		if err != nil {
+			return fmt.Errorf("parse forward ports timeout: %w", err)
+		}
 
-		// start ssh tunnel
-		return cmd.startTunnel(ctx, devPodConfig, containerClient, client.WorkspaceConfig().IDE.Name, log)
-	})
+		log.Infof("Using port forwarding timeout of %s", cmd.ForwardPortsTimeout)
+	}
```

Comment:
move parsing of timeout out to separate function

Reply:
Done :+1: